api/envoy/api/v2/route/route.proto

```diff
@@ -609,6 +609,10 @@ message RouteAction {
   CorsPolicy cors = 17;
 
   reserved 21;
+
+  // When the request is a gRPC request, use the 'grpc-timeout' header, or it's default value
```

Comment:
Yes. That is the defined behavior for missing 'grpc-timeout' header. That is also why I felt this behavior needs to be opt-in via the new config option, as it otherwise would change behavior of existing configs.

Reply:
Before diving into full code review, I have a general question about this: I feel that in general this header should be sanitized for external requests like we do for other similar headers. If we don't wan't that in all cases, I think we might need another option to control this. Thoughts?